File: godtools-db/src/main/java/org/keynote/godtools/android/db/TranslationFileMapper.java
package org.keynote.godtools.android.db;

import android.content.ContentValues;
import android.database.Cursor;
import android.support.annotation.NonNull;

import org.ccci.gto.android.common.db.AbstractMapper;
import org.cru.godtools.model.LocalFile;
import org.cru.godtools.model.Translation;
import org.cru.godtools.model.TranslationFile;

import static org.keynote.godtools.android.db.Contract.TranslationFileTable.COLUMN_FILE;
import static org.keynote.godtools.android.db.Contract.TranslationFileTable.COLUMN_TRANSLATION;

final class TranslationFileMapper extends AbstractMapper<TranslationFile> {
    @Override
    protected void mapField(@NonNull final ContentValues values, @NonNull final String field,
                            @NonNull final TranslationFile file) {
        switch (field) {
            case COLUMN_TRANSLATION:
                values.put(field, file.getTranslationId());
                break;
            case COLUMN_FILE:
                values.put(field, file.getFileName());
                break;
            default:
                super.mapField(values, field, file);
                break;
        }
    }

    @NonNull
    @Override
    protected TranslationFile newObject(@NonNull final Cursor c) {
        return new TranslationFile();
    }

    @NonNull
    @Override
    public TranslationFile toObject(@NonNull final Cursor c) {
        final TranslationFile file = super.toObject(c);

        file.setTranslationId(getLong(c, COLUMN_TRANSLATION, Translation.INVALID_ID));
        file.setFileName(getString(c, COLUMN_FILE, LocalFile.INVALID_FILE_NAME));

        return file;
    }
}

File: library/db/src/main/java/org/keynote/godtools/android/db/TranslationFileMapper.kt
package org.keynote.godtools.android.db

import android.content.ContentValues
import android.database.Cursor
import org.ccci.gto.android.common.db.AbstractMapper
import org.ccci.gto.android.common.util.database.getLong
import org.ccci.gto.android.common.util.database.getString
import org.cru.godtools.model.Base
import org.cru.godtools.model.TranslationFile
import org.keynote.godtools.android.db.Contract.TranslationFileTable.COLUMN_FILE
import org.keynote.godtools.android.db.Contract.TranslationFileTable.COLUMN_TRANSLATION

internal object TranslationFileMapper : AbstractMapper<TranslationFile>() {
    override fun mapField(values: ContentValues, field: String, file: TranslationFile) {
        when (field) {
            COLUMN_TRANSLATION -> values.put(field, file.translationId)
            COLUMN_FILE -> values.put(field, file.filename)
            else -> super.mapField(values, field, file)
        }
    }

    override fun newObject(c: Cursor) = TranslationFile(
        translationId = c.getLong(COLUMN_TRANSLATION, Base.INVALID_ID),
        filename = c.getString(COLUMN_FILE)
    )
}
